### PR TITLE
9.0 attachment preview gs

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please sign the CLA.

File: attachment_preview/README.rst [hunk 1]
```diff
@@ -9,16 +9,30 @@ Installation
...
 =====
 
 The module adds a little print preview icon right of download links for
 attachments or binary fields.
 
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/118/8.0
+
+For further information, please visit:
```
Comment (pedrobaeza) [inline attachment_preview/README.rst:24]: Remove this part

File: attachment_preview/README.rst [hunk 2]
```diff
@@ -9,16 +9,30 @@ Installation
...
 
 Usage
 =====
 
 The module adds a little print preview icon right of download links for
 attachments or binary fields.
 
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+    :alt: Try me on Runbot
+    :target: https://runbot.odoo-community.org/runbot/118/8.0
```
Comment (pedrobaeza) [inline attachment_preview/README.rst:22]: 9.0

**Review by pedrobaeza — APPROVED** [review]
Little comment, but code review + try on runbot correct. Most of the changes are from the new version library.

**Comment by pedrobaeza** [comment]
Please squash all Transbot commits together and all your commits also.

File: attachment_preview/static/src/css/attachment_preview.css [hunk 3]
```diff
@@ -1,11 +1,16 @@
-.openerp .oe_sidebar .oe_dropdown_menu li .oe-sidebar-attachment-preview
+/*.openerp .oe_sidebar .oe_dropdown_menu li .oe_sidebar_attachment_preview
```
Comment (hbrunn) [inline attachment_preview/static/src/css/attachment_preview.css:1]: don't comment out stuff, remove if not necessary

File: attachment_preview/static/src/js/attachment_preview.js [hunk 4]
```diff
@@ -19,41 +19,47 @@
...
+
+    Sidebar.include({
+        on_attachments_loaded: function(attachments) {
+            this._super(attachments);
+
+            this.$('.oe_sidebar_attachment_preview')
                 .click(this.on_attachment_preview);
             this.update_preview_buttons();
-            return result;
+           // return _super.apply(self, arguments);
```
Comment (hbrunn) [inline attachment_preview/static/src/js/attachment_preview.js:59]: delete

File: attachment_preview/static/src/js/attachment_preview.js [hunk 5]
```diff
@@ -208,22 +211,21 @@ openerp.attachment_preview = function(instance)
                 {
                     _(extensions).each(function(extension)
                     {
-                        var $element = self.$el.find('.oe-binary-preview');
-                        if(openerp.attachment_preview.can_preview(extension))
+                        var $element = self.$('.oe_binary_preview');
+                        $element.unbind( "click" );
+                       if(can_preview(extension[0]))
```
Comment (hbrunn) [inline attachment_preview/static/src/js/attachment_preview.js:216]: formatting

File: attachment_preview/static/src/js/attachment_preview.js [hunk 6]
```diff
@@ -208,22 +211,21 @@ openerp.attachment_preview = function(instance)
...
                                     null,
-                                    _.str.sprintf(
-                                        '/web/binary/saveas?session_id=%s&model=%s&field=%s&id=%d',
-                                        instance.session.session_id,
-                                        self.view.dataset.model,
-                                        self.name,
-                                        self.view.datarecord.id),
-                                    extension,
+
+                                        ('data:application/'+extension[0]+';base64,'+
```
Comment (hbrunn) [inline attachment_preview/static/src/js/attachment_preview.js:223]: why would you want to do this? With a few really big attachments, your browser becomes really inresponsive

File: attachment_preview/static/lib/ViewerJS/index.html [hunk 7]
```diff
@@ -39,43 +39,41 @@
...
-b="odt fodt ott odp fodp otp ods fods ots".split(" ");return{supportsMimetype:function(b){return-1!==a.indexOf(b)},supportsFileExtension:function(a){return-1!==b.indexOf(a)},path:"./ODFViewerPlugin",getClass:function(){return ODFViewerPlugin}}}(),{supportsMimetype:function(a){return"application/pdf"===a},supportsFileExtension:function(a){return"pdf"===a},path:"./PDFViewerPlugin",getClass:function(){return PDFViewerPlugin}}];window.onload=function(){var a=document.location.hash.substring(1),d=r(document.location),
-c;a?(d.title||(d.title=a.replace(/^.*[\\\/]/,"")),d.documentUrl=a,b(a,function(b){b||(b=d.type?f(d.type):x(a));b?"undefined"!==String(typeof loadPlugin)?loadPlugin(b.path,function(){c=b.getClass();new Viewer(new c,d)}):(c=b.getClass(),new Viewer(new c,d)):new Viewer})):new Viewer};k=document.createElementNS(document.head.namespaceURI,"style");k.setAttribute("media","screen");k.setAttribute("type","text/css");k.appendChild(document.createTextNode(viewer_css));document.head.appendChild(k);k=document.createElementNS(document.head.namespaceURI,
+m.webkitReques
```
Comment (hbrunn) [inline attachment_preview/static/lib/ViewerJS/index.html:63]: if this is not needed any more, also delete the file

File: attachment_preview/static/src/js/viewerjs_tweaks.js [hunk 8]
```diff
@@ -24,10 +24,9 @@
...
-    var matches = (/&title=([^&]+)&/).exec(window.location.hash);
-    if(matches && matches.length > 1)
+    if(!plugin)
     {
-        parameters.title = decodeURIComponent(matches[1]);
+        alert('Unsupported file type');
     }
     return original_Viewer(plugin, parameters);
-}
+};
```
Comment (hbrunn) [inline attachment_preview/static/src/js/viewerjs_tweaks.js:32]: you don't include this file any more, so it can go away i guess?

**Review by hbrunn — CHANGES_REQUESTED** [review]

**Comment by hbrunn** [comment]
adding tests to have coverage at least not drop would be good

**Pull request closed without merging**